engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/typer/TypeMethodReference.scala

```diff
@@ -7,19 +7,22 @@ import pl.touk.nussknacker.engine.definition.TypeInfos.{ClazzDefinition, MethodI
 import pl.touk.nussknacker.engine.types.EspTypeUtils
 
 object TypeMethodReference {
-  def apply(methodName: String, currentResults: List[TypingResult], params: List[TypingResult], typeDefinitionSet: TypeDefinitionSet)(implicit settings: ClassExtractionSettings): Either[String, TypingResult] =
-    new TypeMethodReference(methodName, currentResults, params, typeDefinitionSet).call
+  def apply(methodName: String, currentResults: List[TypingResult], params: List[TypingResult], typeDefinitionSet: TypeDefinitionSet, disableMethodExecutionForUnknown: Boolean)(implicit settings: ClassExtractionSettings): Either[String, TypingResult] =
+    new TypeMethodReference(methodName, currentResults, params, typeDefinitionSet, disableMethodExecutionForUnknown).call
 }
 
-class TypeMethodReference(methodName: String, currentResults: List[TypingResult], calledParams: List[TypingResult], typeDefinitionSet: TypeDefinitionSet) {
+class TypeMethodReference(methodName: String, currentResults: List[TypingResult], calledParams: List[TypingResult], typeDefinitionSet: TypeDefinitionSet, disableMethodExecutionForUnknown: Boolean) {
   def call(implicit settings: ClassExtractionSettings): Either[String, TypingResult] =
     currentResults.headOption match {
       case Some(tc: SingleTypingResult) =>
         typeFromClazzDefinitions(extractClazzDefinitions(Set(tc)))
       case Some(TypedUnion(nestedTypes)) =>
         typeFromClazzDefinitions(extractClazzDefinitions(nestedTypes))
       case _ =>
-        Right(Unknown)
+        if(disableMethodExecutionForUnknown)
```

Comment:
Can Unknown be returned from above cases ?

Reply:
:+1: Good catch. Let's do the following:
- replace `currentResults: List[TypingResult]` with `invocationTarget: TypingResult`, as we already check that list is not empty in `Typer::extractMethodReference`
- add explicit `case Unknown` here

Reply:
Alright, things changed.

Reply:
Alright, things changed.